src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Completion/Delegation/DelegatedCompletionListProviderTest.cs

```diff
@@ -222,6 +228,65 @@ public async Task CSharpDelegation_UnsupportedTriggerCharacter_TranslatesToInvok
             Assert.Equal(CompletionTriggerKind.Invoked, delegatedParameters.Context.TriggerKind);
             Assert.Equal(VSInternalCompletionInvokeKind.Typing, delegatedParameters.Context.InvokeKind);
             Assert.Equal(1337, delegatedParameters.HostDocument.Version);
+            Assert.Null(delegatedParameters.ProvisionalTextEdit);
+        }
+
+        [Fact]
+        public async Task ProvisionalCompletion_TranslatesToCSharpWithProvisionalTextEdit()
+        {
+            // Arrange
+            var completionContext = new VSInternalCompletionContext()
+            {
+                InvokeKind = VSInternalCompletionInvokeKind.Typing,
+                TriggerKind = CompletionTriggerKind.TriggerCharacter,
+                TriggerCharacter = ".",
+            };
+            var codeDocument = CreateCodeDocument("@DateTime.");
+            var documentContext = TestDocumentContext.From("C:/path/to/file.cshtml", codeDocument, hostDocumentVersion: 1337);
+
+            // Act
+            await Provider.GetCompletionListAsync(absoluteIndex: 10, completionContext, documentContext, ClientCapabilities, CancellationToken.None);
+
+            // Assert
+            var delegatedParameters = Provider.DelegatedParams;
+            Assert.NotNull(delegatedParameters);
+            Assert.Equal(RazorLanguageKind.CSharp, delegatedParameters.ProjectedKind);
+
+            // Just validating that we're generating code in a way that's different from the top-level document. Don't need to be specific.
+            Assert.True(delegatedParameters.ProjectedPosition.Line > 2);
+            Assert.Equal(CompletionTriggerKind.TriggerCharacter, delegatedParameters.Context.TriggerKind);
+            Assert.Equal(VSInternalCompletionInvokeKind.Typing, delegatedParameters.Context.InvokeKind);
+            Assert.Equal(1337, delegatedParameters.HostDocument.Version);
+            Assert.NotNull(delegatedParameters.ProvisionalTextEdit);
```

Comment:
Maybe we should verify the contents of the edit?

Reply:
I thought about that; however, the edit itself isn't important for this test. It's more about "is there one"